File: solutions/julia/armstrong-numbers/1/armstrong-numbers.jl
function isarmstrong(n)
    ds = digits(n)
    l = length(ds)
    sum(ds .^ l) == n
end


function digits(n)
    ds = Int64[]
    while n >= 10
        push!(ds, n % 10)
        n = div(n, 10)
    end
    push!(ds, n)
end

File: solutions/haskell/armstrong-numbers/1/src/ArmstrongNumbers.hs
module ArmstrongNumbers (armstrong) where

armstrong :: Integral a => a -> Bool
armstrong n =
  let
    ds = digits n
    armSum = sum $ map (^(length ds)) ds
  in
    armSum == n

digits :: Integral a => a -> [a]
digits n =
  if n < 10 then 
    [n]
  else
    (n `mod` 10):(digits (n `div` 10))
